tests/contracts/TestRulesPaymaster.sol

```diff
@@ -30,12 +34,20 @@ contract TestRulesPaymaster is IPaymaster, ValidationRulesStorage {
 
         //first byte after paymaster address.
         string memory rule = string(userOp.paymasterAndData[UserOperationLib.PAYMASTER_DATA_OFFSET:]);
-        if (rule.eq("context")) {
+        if (rule.includes("context")) {
```

Comment:
i must have missed something: why ignore the full string, and use "includes" ? if a prefix was matched earlier, then it should be stripped, so inner check should still do "eq"

Reply:
Well, `"context"` and `"nothing"` are not really "rules" in the normal sense, they are keywords for the Paymaster to do something different. (You can notice that the `runRule` is callled in the "else" clause here.)
So I had to work around this abuse of the `rule` variable if I didn't want to rewrite the entire contract.
I still think it should be rewritten, just not as a part of this PR which is big enough and should focus on testing the tracer/parser.